src/core/domain.py

```diff
@@ -259,3 +266,178 @@ def zookeeper_connected(self) -> bool:
     def __bool__(self) -> bool:
         """Return whether this class object has sufficient information."""
         return self.zookeeper_connected
+
+
+@dataclass
+class TLSInfo:
+    """Class representing a information related to tls."""
+
+    keystore_password: str
+    trustore_password: str
+
+
+@dataclass
+class SANs:
+    """Subject Alternative Name (SAN)s used to create multi-domains certificates."""
+
+    sans_ip: list[str]
+    sans_dns: list[str]
+
+
+class KyuubiServer(RelationState):
+    """State collection metadata for a charm unit."""
+
+    def __init__(
+        self,
+        relation: Relation | None,
+        data_interface: Data,
+        component: Unit,
+    ):
+        super().__init__(relation, data_interface, component)
+        self.unit = component
+        self.k8s = ServiceManager(
+            self.unit._backend.model_name, self.unit.name, self.unit.app.name
+        )
+
+    @property
+    def unit_id(self) -> int:
+        """The id of the unit from the unit name.
+
+        e.g kyuubi/2 --> 2
+        """
+        return int(self.unit.name.split("/")[1])
+
+    @property
+    def hostname(self) -> str:
+        """The hostname for the unit."""
+        return self.relation_data.get("hostname", "")
+
+    @property
+    def fqdn(self) -> str:
+        """The Fully Qualified Domain Name for the unit."""
+        return self.relation_data.get("fqdn", "")
+
+    @property
+    def ip(self) -> str:
+        """The IP for the unit."""
+        return self.relation_data.get("ip", "")
+
+    @property
+    def host(self) -> str:
+        """The hostname for the unit."""
+        return f"{self.unit.name.split('/')[0]}-{self.unit_id}.{self.unit.name.split('/')[0]}-endpoints"
+
+    # -- TLS --
+
+    @property
+    def private_key(self) -> str:
+        """The private-key contents for the unit to use for TLS."""
+        return self.relation_data.get("private-key", "")
+
+    @property
+    def keystore_password(self) -> str:
+        """The Java Keystore password for the unit to use for TLS."""
+        return self.relation_data.get("keystore-password", "")
+
+    @property
+    def truststore_password(self) -> str:
+        """The Java Truststore password for the unit to use for TLS."""
+        return self.relation_data.get("truststore-password", "")
+
+    @property
+    def csr(self) -> str:
+        """The current certificate signing request contents for the unit."""
+        return self.relation_data.get("csr", "")
+
+    @property
+    def certificate(self) -> str:
+        """The certificate contents for the unit to use for TLS."""
+        return self.relation_data.get("certificate", "")
+
+    @property
+    def ca(self) -> str:
+        """The root CA contents for the unit to use for TLS."""
+        # Backwards compatibility
+        # TODO (zkclient): Remove this property and replace by "" in self.ca_cert
```

Comment:
Do we need this backward compatibility? It seems to me that this is a leftover from Zookeeper code.